### DISCUSSION TITLE
Lazy loading collapsible table

### DISCUSSION THREAD

**Discussion by SebastiaanKloos** [filamentphp/filament#5692] [Ideas]
It would be really cool if you can lazy load the content in the collapsed view.

https://filamentphp.com/docs/2.x/tables/layout#collapsible-content

**Comment by krizzdev** [comment]
This is a feature I am looking forward to.

We have a lot of tables in our project where we want to display a quick view (collapse) and show more information about some relational data. 
Eager loading all of this would slow down the table pagination and loading experience.
It would be a cool feature to load these on purpose (onclick)

**Comment by mrvnklm** [comment]
any news or workaround on this?

**Comment by LloydLabs** [comment]
I'm also looking for a solution around this

**Comment by SuperUserDo-Nikola** [comment]
Hi, I have done all custom.

I created custom let's call it "expandable-column" which is coming from custom filament columns that are included into table() function in resource or page or livewire component and added JS with Alpine click handlers into that class which will trigger and populate html with content I provide. I this case main thing is I provided filament "tabs" into view and this is end result I got. 

Content that is expandable is lazy loaded so it is perfect for large datasets which I had issue with. 

Thing to note is that Main tabs is from filament, but each tab is Livewire component which is also lazy loaded for performance 😅

<img width="1410" alt="image" src="https://github.com/user-attachments/assets/3d1c3d42-becf-464c-8b60-89659e2cdd07" />
<img width="1528" alt="image" src="https://github.com/user-attachments/assets/1c9967c8-3b2f-4b36-b2dc-af47bd4a7ef5" />

**Reply by PasqualeBassolillo** [reply]
@SuperUserDo-Nikola Can I ask you if you can make that custom column open-source?

**Reply by skaesdorf** [reply]
@SuperUserDo-Nikola could you please share the code of this column? I am very intereseted in what you did there. Thanks!

**Reply by bjornvb** [reply]
@SuperUserDo-Nikola I'am also interested in how you did this. Thanks

**Reply by SuperUserDo-Nikola** [reply]
Here you go fellas, public git repo -> https://github.com/SuperUserDo-Nikola/filament-4-table-collapsible-column

It is missing README, did not have time for it. But you can clone project, it it basic Laravel, Filament app. Run Database seeder for creation of User (needed for Auth). 

Once you are logged in, go to user resource and enjoy. In Expandable column you can put pretty much everything 😅
